pinot-common/src/main/java/org/apache/pinot/common/utils/config/TableConfigUtils.java

```diff
@@ -444,4 +447,36 @@ public static boolean hasPreConfiguredInstancePartitions(TableConfig tableConfig
     return hasPreConfiguredInstancePartitions(tableConfig)
         && tableConfig.getInstancePartitionsMap().containsKey(instancePartitionsType);
   }
+
+  /**
+   * Get the partition column from InstanceAssignmentConfigUtils
+   * @param tableConfig table config
+   * @return partition column
+   */
+  public static String getPartitionColumn(TableConfig tableConfig) {
+    String partitionColumn = null;
+
+    // check getInstanceAssignmentConfigMap is null or empty,
+    if (!MapUtils.isEmpty(tableConfig.getInstanceAssignmentConfigMap())) {
+      for (String key : tableConfig.getInstanceAssignmentConfigMap().keySet()) {
+        //check getInstanceAssignmentConfigMap has the key of TableType
+        if (Objects.equals(key, tableConfig.getTableType().toString())) {
```

Comment:
The key of the map is not always the table type. We can simply loop over the map (`values()`), and return the partition column if it is configured

Reply:
I have made the changes. Thanks